source/Octopus.Server.Client/Repositories/Async/BasicRepository.cs

```diff
@@ -253,6 +253,26 @@ public async Task<List<TResource>> GetAll(CancellationToken cancellationToken)
             return await Client.Get<List<TResource>>(link, parameters, cancellationToken).ConfigureAwait(false);
         }
 
+        public Task<List<TResource>> FindByPartialName(string partialName, string path, object pathParameters, CancellationToken cancellationToken)
+        {
+            ThrowIfServerVersionIsNotCompatible(cancellationToken).ConfigureAwait(false);
+
+            partialName = (partialName ?? string.Empty).Trim();
+            if (pathParameters == null)
+                pathParameters = new { partialName = partialName};
+
+            return FindMany(r =>
+            {
+                var named = r as INamedResource;
+                return named != null && named.Name.Contains(partialName);
+            }, path, pathParameters);
```

Comment:
It looks like similar methods like `FindByName` are case insensitive, but this is case sensitive. It'd be good to keep them the same. I think you'll need to `IndexOf` because we're on `netstandard2.0`, the case insensitive `Contains` overload wasn't added til later IIRC:

```suggestion
            return FindMany(
                r => r is INamedResource named && named.Name.IndexOf(partialName, StringComparison.OrdinalIgnoreCase) != -1,
                path,
                pathParameters);
```

Might be worth double checking that the tenants endpoint is case insensitive too.

Reply:
Thanks Ross! it looks like we do use .contains in other places

https://github.com/OctopusDeploy/OctopusClients/blob/b883446b1286904f102ca46013eee4927f02a967/source/Octopus.Server.Client/Repositories/Async/AccountRepository.cs#L76

https://github.com/OctopusDeploy/OctopusClients/blob/0a9f0eb5ecf9a9b064135ae4233bdd57544cfb5b/source/Octopus.Server.Client/OctopusAsyncRepository.cs#L218

Would IndexOf still be preferred? 

Good catch on the case sensitivity, I'll take a look

Reply:
Actually all good, was easy enough to add in an extension where you can pass in the OrdinalIgnoreCase

Reply:
`Contains` is preferred where it's possible, it's just that `Contains(string, StringComparison)` wasn't added til .NET Standard 2.1 so it's the only option unless we upgrade. Your extension method is a solid approach 😄